src/OpenFeature/Providers/Memory/InMemoryProvider.cs

```diff
@@ -0,0 +1,137 @@
+        /// </summary>
+        /// <param name="flags">the flags to use instead of the previous flags.</param>
+        public async ValueTask UpdateFlags(IDictionary<string, Flag>? flags = null)
+        {
+            var changed = this._flags.Keys.ToList();
+            if (flags == null)
+            {
+                this._flags = new Dictionary<string, Flag>();
+            }
+            else
+            {
+                this._flags = new Dictionary<string, Flag>(flags); // shallow copy
+            }
+            changed.AddRange(this._flags.Keys.ToList());
+            var @event = new ProviderEventPayload
+            {
+                Type = ProviderEventTypes.ProviderConfigurationChanged,
+                ProviderName = _metadata.Name,
+                FlagsChanged = changed, // emit all
+                Message = "flags changed",
+            };
+            await this.EventChannel.Writer.WriteAsync(@event).ConfigureAwait(false);
+        }
+
+        /// <inheritdoc/>
+        public override Task<ResolutionDetails<bool>> ResolveBooleanValue(
+            string flagKey,
+            bool defaultValue,
+            EvaluationContext? context = null)
+        {
+            return Task.FromResult(Resolve(flagKey, defaultValue, context));
+        }
+
+        /// <inheritdoc/>
+        public override Task<ResolutionDetails<string>> ResolveStringValue(
+            string flagKey,
+            string defaultValue,
+            EvaluationContext? context = null)
+        {
+            return Task.FromResult(Resolve(flagKey, defaultValue, context));
+        }
+
+        /// <inheritdoc/>
+        public override Task<ResolutionDetails<int>> ResolveIntegerValue(
+            string flagKey,
+            int defaultValue,
+            EvaluationContext? context = null)
+        {
+            return Task.FromResult(Resolve(flagKey, defaultValue, context));
+        }
+
+        /// <inheritdoc/>
+        public override Task<ResolutionDetails<double>> ResolveDoubleValue(
+            string flagKey,
+            double defaultValue,
+            EvaluationContext? context = null)
+        {
+            return Task.FromResult(Resolve(flagKey, defaultValue, context));
+        }
+
+        /// <inheritdoc/>
+        public override Task<ResolutionDetails<Value>> ResolveStructureValue(
+            string flagKey,
+            Value defaultValue,
+            EvaluationContext? context = null)
+        {
+            return Task.FromResult(Resolve(flagKey, defaultValue, context));
+        }
+
+        private ResolutionDetails<T> Resolve<T>(string flagKey, T defaultValue, EvaluationContext? context)
+        {
+            if (!this._flags.TryGetValue(flagKey, out var flag))
+            {
+                throw new FlagNotFoundException($"flag {flag} not found");
+            }
+            else
+            {
+                if (typeof(Flag<T>).Equals(flag.GetType()))
```

Comment:
So, this is maybe fine for an in-memory provider, but in general it gives me pause. In many typical cases the underlying transport has "numbers". So JSON, for instance, everything is basically a double. People then ask for that double as an integer or double.

Reply:
A person who is using this as a short-cut to an in-house provider will just need to be mindful.

Reply:
Great point. I've seen this issue elsewhere. Any suggestions? It's possible people will look at this implementation as a reference. Is there anything that can be done that's more elegant than checking specifically if these are any kind of numbers and casting?

Reply:
I am not sure of a simple way to handle it. The way we handle it in LD SDKs is basically to always store doubles and then the int/double method use both use double. The int method truncates. Our data type is from JSON, so we don't lose fidelity compared to the source. Someone who actually supports ints this would be a limitation.

I am not sure I see an easy way to handle it here other than to have special handling for numbers.

Get the flag, check if it is the requested numeric type, if it isn't check if it is the other one, then do the conversion. Maybe it could be somewhat clean with switching on the type.

Reply:
If you agree it's sufficient, I think I'll just add a comment explaining the shortcoming of this check, since the in-memory provider is is not a production tool. This will at least be edifying for anyone looking at the implementation for reference.

Reply:
done:

```
// This check returns False if a floating point flag is evaluated as an integer flag, and vice-versa.
// In a production provider, such behavior is probably not desirable; consider supporting conversion.
```